int16 rb_fr
int16 rb_mr
int16 rb_rr
int16 rb_fl
int16 rb_ml
int16 rb_rl
